doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,399 @@
+
+The `defaults.json` file will be written out whenever we detect that it doesn't exist.
+
+**TODO**: We also need to write it out if it changes, though ideally without
+actually loading the file at first load. Perhaps we could start another
+thread/`fire_and_forget` once we're loaded to re-scan the file and re-write it
+if it has changed.
+
+Because the `defaults.json` file is hardcoded within our application, we can use
+its text directly, without loading the file from disk. This should help save
+some startup time, as we'll only need to load the user settings from disk.
+
+When we make changes to the default settings, or we make changes to the settings
+schema, we should make sure that we update the hardcoded `defaults.json` with
+the new values. That way, the `defaults.json` file will always have the complete
+set of setings in it.
+
+
+### Layering settings
+
+When we load the settings, we'll do it in two stages. First, we'll deseriale the
+default settings that we've hardcoded. Then, we'll intelligently layer the
+user's setting upon those we've already loaded. Some objects, like the default
+profiles, we'll need to make sure to load from the user settings into the
+existing objects we created from the default settings.
+
+* We'll need to make sure that any profile in the user settings that has a GUID
+  matching a default profile loads the user settings into the object created
+  from the defaults.
+* We'll need to make sure that there's only one action bound to each key chord
+  for a keybinding. If there are any key chords in the user settings that match
+  a default key chord, we should bind them to the action from the user settings
+  instead.
+* For any color schemes who's name matches the name of a default color scheme,
+  we'll need to apply the user settings to the existing color scheme.
+
+### Hiding Default Profiles
+
+What if a user doesn't want to see one of the profiles that we've included in
+the default profiles?
+
+We will add a `hidden` key to each profile, which defaults to false. When we
+want to mark a profile as hidden, we'd just set that value to `true`, instead of
+trying to look up the profile's guid.
+
+
+So, if someone wanted to hide the default cmd.exe profile, all they'd have to do
+is add `"hidden": true` to the cmd.exe entry in their user settings, like so:
+
+```js
+{
+    "profiles": [
+        {
+            // Make changes here to the cmd.exe profile
+            "guid": "{6239a42c-1de4-49a3-80bd-e8fdd045185c}",
+            "hidden": true
+        }
+    ],
+```
+
+#### Hidden Profiles and the Open New Tab shortcuts
+
+Currently, there are keyboard shortcuts for "Open New Tab With Profile
+&lt;N&gt;". These shortcuts will open up the Nth profile in the new tab
+dropdown. Considering we're adding the ability to remove profiles from that
+list, but keep them in the overall list of profiles, we'll need to make sure
+that the handler for that event still opens the Nth _visible_ profile.

```

Comment:
On that note, can the user reorder the profiles by changing the order of _their_ guid blocks?

Reply:
This is a good point. We'll have to deal with that.